### PR TITLE
timex: make timex_to_str more efficient

### PR DESCRIPTION
**Pull request by authmillenon**
Attempt to fulfill the marked TODO:

```
text	data	bss	dec	BOARD/BINDIRBASE

-660	0	0	-660	airfy-beacon
9564	112	2688	12364	master-bin
8904	112	2688	11704	timex_to_str-bin

-644	0	0	-644	arduino-due
9852	116	2764	12732	master-bin
9208	116	2764	12088	timex_to_str-bin

124	-12	0	112	arduino-mega2560
8566	334	770	9670	master-bin
8690	322	770	9782	timex_to_str-bin

-8408	0	0	-8408	avsextrem
48516	2232	96069	146817	master-bin
40108	2232	96069	138409	timex_to_str-bin

-644	0	0	-644	cc2538dk
9364	112	2688	12164	master-bin
8720	112	2688	11520	timex_to_str-bin

-110	0	-4	-114	chronos
8762	90	1060	9912	master-bin
8652	90	1056	9798	timex_to_str-bin

-640	0	0	-640	ek-lm4f120xl
9380	112	2728	12220	master-bin
8740	112	2728	11580	timex_to_str-bin

-640	0	0	-640	f4vi1
10880	116	2700	13696	master-bin
10240	116	2700	13056	timex_to_str-bin

-644	0	0	-644	fox
11184	112	2704	14000	master-bin
10540	112	2704	13356	timex_to_str-bin

-644	0	0	-644	frdm-k64f
9328	1140	4820	15288	master-bin
8684	1140	4820	14644	timex_to_str-bin

-644	0	0	-644	iot-lab_M3
11212	112	2704	14028	master-bin
10568	112	2704	13384	timex_to_str-bin

-644	0	0	-644	mbed_lpc1768
9184	112	2712	12008	master-bin
8540	112	2712	11364	timex_to_str-bin

56	0	-4	52	msb-430
7996	10	1090	9096	master-bin
8052	10	1086	9148	timex_to_str-bin

56	0	-4	52	msb-430h
7874	10	1090	8974	master-bin
7930	10	1086	9026	timex_to_str-bin

-8408	0	0	-8408	msba2
48172	2232	96069	146473	master-bin
39764	2232	96069	138065	timex_to_str-bin

-644	0	0	-644	msbiot
11164	116	2724	14004	master-bin
10520	116	2724	13360	timex_to_str-bin

-644	0	0	-644	mulle
11380	1164	4564	17108	master-bin
10736	1164	4564	16464	timex_to_str-bin

0	0	0	0	native
25953	392	51752	78097	master-bin
25953	392	51752	78097	timex_to_str-bin

-660	0	0	-660	nrf51dongle
9588	112	2688	12388	master-bin
8928	112	2688	11728	timex_to_str-bin

-664	0	0	-664	nucleo-f091
12748	112	2712	15572	master-bin
12084	112	2712	14908	timex_to_str-bin

-644	0	0	-644	nucleo-f303
10840	112	2720	13672	master-bin
10196	112	2720	13028	timex_to_str-bin

-640	0	0	-640	nucleo-f334
10580	112	2696	13388	master-bin
9940	112	2696	12748	timex_to_str-bin

-644	0	0	-644	nucleo-l1
10792	112	2704	13608	master-bin
10148	112	2704	12964	timex_to_str-bin

-644	0	0	-644	openmote
9316	112	2688	12116	master-bin
8672	112	2688	11472	timex_to_str-bin

-644	0	0	-644	pba-d-01-kw2x
9396	1140	4204	14740	master-bin
8752	1140	4204	14096	timex_to_str-bin

-660	0	0	-660	pca10000
9588	112	2688	12388	master-bin
8928	112	2688	11728	timex_to_str-bin

-612	0	0	-612	pca10005
9568	112	2688	12368	master-bin
8956	112	2688	11756	timex_to_str-bin

-8408	0	0	-8408	pttu
48724	2232	96069	147025	master-bin
40316	2232	96069	138617	timex_to_str-bin

56	0	0	56	qemu-i386
128651	5464	79508	213623	master-bin
128707	5464	79508	213679	timex_to_str-bin

-8456	0	0	-8456	redbee-econotag
49728	2264	8972	60964	master-bin
41272	2264	8972	52508	timex_to_str-bin

-660	0	0	-660	saml21-xpro
9568	112	2792	12472	master-bin
8908	112	2792	11812	timex_to_str-bin

-660	0	0	-660	samr21-xpro
9736	112	2680	12528	master-bin
9076	112	2680	11868	timex_to_str-bin

-644	0	0	-644	spark-core
11240	112	2704	14056	master-bin
10596	112	2704	13412	timex_to_str-bin

-664	0	0	-664	stm32f0discovery
12752	112	2712	15576	master-bin
12088	112	2712	14912	timex_to_str-bin

-644	0	0	-644	stm32f3discovery
10844	112	2720	13676	master-bin
10200	112	2720	13032	timex_to_str-bin

-640	0	0	-640	stm32f4discovery
11044	116	2716	13876	master-bin
10404	116	2716	13236	timex_to_str-bin

56	0	-4	52	telosb
7936	6	1090	9032	master-bin
7992	6	1086	9084	timex_to_str-bin

-644	0	0	-644	udoo
9852	116	2764	12732	master-bin
9208	116	2764	12088	timex_to_str-bin

56	0	-4	52	wsn430-v1_3b
8026	10	1090	9126	master-bin
8082	10	1086	9178	timex_to_str-bin

56	0	-4	52	wsn430-v1_4
8026	10	1090	9126	master-bin
8082	10	1086	9178	timex_to_str-bin

-660	0	0	-660	yunjia-nrf51822
9552	112	2688	12352	master-bin
8892	112	2688	11692	timex_to_str-bin

36	0	-4	32	z1
7656	6	1090	8752	master-bin
7692	6	1086	8784	timex_to_str-bin
```

Since `timex_to_str()` is currently used nowhere I used a [tailor-made application](https://gist.github.com/authmillenon/b246a473811791384bcc).

I can't explain why it is bigger on some platforms (especially qemu which should be 0 as native is).

I also tried to implement itoa for the platforms it is missing, but it got bigger than the current solution

### CODE REVIEW

**Comment by authmillenon** [comment]
While writing unittests for this I realized that this won't work for `((int)t.seconds) < 0` since `itoa()` is expecting signed integers and converts them accordingly. Will offer a fix and the size difference of that shortly.

**Comment by authmillenon** [comment]
First the fix.

**Comment by authmillenon** [comment]
Now diff. This time I set `-DLOG_LEVEL=LOG_NONE` to get at least the kernel to not pull in `printf()`. There seem to be still some usage of `printf()` by some boards/cpus that skew the results a little, but for those that don't we win ~800 byte with nano-specs and ~

```
text    data    bss     dec     BOARD/BINDIRBASE

-780    0       0       -780    airfy-beacon
9564    112     2688    12364   master-bin
8784    112     2688    11584   timex_to_str-bin

-804    0       0       -804    arduino-due
9852    116     2764    12732   master-bin
9048    116     2764    11928   timex_to_str-bin

194     -12     0       182     arduino-mega2560
8566    334     770     9670    master-bin
8760    322     770     9852    timex_to_str-bin

-8584   0       0       -8584   avsextrem
48516   2232    96069   146817  master-bin
39932   2232    96069   138233  timex_to_str-bin

-804    0       0       -804    cc2538dk
9364    112     2688    12164   master-bin
8560    112     2688    11360   timex_to_str-bin

38      0       -4      34      chronos
8762    90      1060    9912    master-bin
8800    90      1056    9946    timex_to_str-bin

-796    0       0       -796    ek-lm4f120xl
9380    112     2728    12220   master-bin
8584    112     2728    11424   timex_to_str-bin

-796    0       0       -796    f4vi1
10880   116     2700    13696   master-bin
10084   116     2700    12900   timex_to_str-bin

-804    0       0       -804    fox
11184   112     2704    14000   master-bin
10380   112     2704    13196   timex_to_str-bin

-800    0       0       -800    frdm-k64f
9328    1140    4820    15288   master-bin
8528    1140    4820    14488   timex_to_str-bin

-804    0       0       -804    iot-lab_M3
11212   112     2704    14028   master-bin
10408   112     2704    13224   timex_to_str-bin

-804    0       0       -804    mbed_lpc1768
9184    112     2712    12008   master-bin
8380    112     2712    11204   timex_to_str-bin

50      0       -4      46      msb-430
7996    10      1090    9096    master-bin
8046    10      1086    9142    timex_to_str-bin

50      0       -4      46      msb-430h
7874    10      1090    8974    master-bin
7924    10      1086    9020    timex_to_str-bin

-8576   0       0       -8576   msba2
48172   2232    96069   146473  master-bin
39596   2232    96069   137897  timex_to_str-bin

-800    0       0       -800    msbiot
11164   116     2724    14004   master-bin
10364   116     2724    13204   timex_to_str-bin

-800    0       0       -800    mulle
11380   1164    4564    17108   master-bin
10580   1164    4564    16308   timex_to_str-bin

350     0       0       350     native
25953   392     51752   78097   master-bin
26303   392     51752   78447   timex_to_str-bin

-780    0       0       -780    nrf51dongle
9588    112     2688    12388   master-bin
8808    112     2688    11608   timex_to_str-bin

-784    0       0       -784    nucleo-f091
12748   112     2712    15572   master-bin
11964   112     2712    14788   timex_to_str-bin

-800    0       0       -800    nucleo-f303
10840   112     2720    13672   master-bin
10040   112     2720    12872   timex_to_str-bin

-796    0       0       -796    nucleo-f334
10580   112     2696    13388   master-bin
9784    112     2696    12592   timex_to_str-bin

-804    0       0       -804    nucleo-l1
10792   112     2704    13608   master-bin
9988    112     2704    12804   timex_to_str-bin

-804    0       0       -804    openmote
9316    112     2688    12116   master-bin
8512    112     2688    11312   timex_to_str-bin

-800    0       0       -800    pba-d-01-kw2x
9396    1140    4204    14740   master-bin
8596    1140    4204    13940   timex_to_str-bin

-780    0       0       -780    pca10000
9588    112     2688    12388   master-bin
8808    112     2688    11608   timex_to_str-bin

-780    0       0       -780    pca10005
9568    112     2688    12368   master-bin
8788    112     2688    11588   timex_to_str-bin

-8584   0       0       -8584   pttu
48724   2232    96069   147025  master-bin
40140   2232    96069   138441  timex_to_str-bin

362     0       0       362     qemu-i386
128651  5464    79508   213623  master-bin
129013  5464    79508   213985  timex_to_str-bin

-8648   0       0       -8648   redbee-econotag
49728   2264    8972    60964   master-bin
41080   2264    8972    52316   timex_to_str-bin

-780    0       0       -780    saml21-xpro
9568    112     2792    12472   master-bin
8788    112     2792    11692   timex_to_str-bin

-780    0       0       -780    samr21-xpro
9736    112     2680    12528   master-bin
8956    112     2680    11748   timex_to_str-bin

-804    0       0       -804    spark-core
11240   112     2704    14056   master-bin
10436   112     2704    13252   timex_to_str-bin

-784    0       0       -784    stm32f0discovery
12752   112     2712    15576   master-bin
11968   112     2712    14792   timex_to_str-bin

-800    0       0       -800    stm32f3discovery
10844   112     2720    13676   master-bin
10044   112     2720    12876   timex_to_str-bin

-796    0       0       -796    stm32f4discovery
11044   116     2716    13876   master-bin
10248   116     2716    13080   timex_to_str-bin

50      0       -4      46      telosb
7936    6       1090    9032    master-bin
7986    6       1086    9078    timex_to_str-bin

-804    0       0       -804    udoo
9852    116     2764    12732   master-bin
9048    116     2764    11928   timex_to_str-bin

50      0       -4      46      wsn430-v1_3b
8026    10      1090    9126    master-bin
8076    10      1086    9172    timex_to_str-bin

50      0       -4      46      wsn430-v1_4
8026    10      1090    9126    master-bin
8076    10      1086    9172    timex_to_str-bin

-780    0       0       -780    yunjia-nrf51822
9552    112     2688    12352   master-bin
8772    112     2688    11572   timex_to_str-bin

38      0       -4      34      z1
7656    6       1090    8752    master-bin
7694    6       1086    8786    timex_to_str-bin
```

**Comment by OlegHahm** [comment]
ACK if Travis agrees. Seems to be at least an improvement.

File: sys/include/timex.h [hunk 1]
```diff
@@ -22,7 +22,6 @@
 #define __TIMEX_H
 
 #include <stdint.h>
-#include <stdio.h>
```
Comment (OlegHahm) [inline sys/include/timex.h]: apparently this removal had some side-effects...

Comment (authmillenon) [inline sys/include/timex.h]: That's unfortunate... Well then let's make this PR a cleanup of other people's messes to, shalt we :-)

**Comment by authmillenon** [comment]
Changed and immediately squashed (and it was my mistake ^^").

File: sys/include/timex.h (hunk 1 above)
Comment (kaspar030) [inline sys/include/timex.h]: Ey.

**Pull request merged**